govcd/lbserverpool.go

```diff
@@ -0,0 +1,194 @@
+	_, err = eGW.client.ExecuteRequest(httpPath, http.MethodGet, types.AnyXMLMime,
+		"unable to read load lalancer server pool: %s", nil, lbPoolResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	// Search for pool by ID or by Name
+	for _, pool := range lbPoolResponse.LBPools {
+		// If ID was specified for lookup - look for the same ID
+		if lbPoolConfig.ID != "" && pool.ID == lbPoolConfig.ID {
+			return pool, nil
+		}
+
+		// If Name was specified for lookup - look for the same Name
+		if lbPoolConfig.Name != "" && pool.Name == lbPoolConfig.Name {
+			// We found it by name. Let's verify if search ID was specified and it matches the lookup object
+			if lbPoolConfig.ID != "" && pool.ID != lbPoolConfig.ID {
+				return nil, fmt.Errorf("load balancer server pool was found by name (%s), but it's ID (%s) does not match specified ID (%s)",
+					pool.Name, pool.ID, lbPoolConfig.ID)
+			}
+			return pool, nil
+		}
+	}
+
+	return nil, fmt.Errorf("could not find load balancer server pool (name: %s, ID: %s)",
+		lbPoolConfig.Name, lbPoolConfig.ID)
+}
+
+// UpdateLBServerPool updates types.LBPool with all fields. At least name or ID must be specified.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+// Name and Algorithm fields must be populated.
+func (eGW *EdgeGateway) UpdateLBServerPool(lbPoolConfig *types.LBPool) (*types.LBPool, error) {
+	if err := validateUpdateLBServerPool(lbPoolConfig); err != nil {
+		return nil, err
+	}
+
+	// if only name was specified for update, ID must be found, because ID is mandatory for update
+	if lbPoolConfig.ID == "" {
+		readLBPool, err := eGW.ReadLBServerPool(&types.LBPool{Name: lbPoolConfig.Name})
+		if err != nil {
+			return nil, fmt.Errorf("unable to find load balancer pool by name for update: %s", err)
+		}
+		lbPoolConfig.ID = readLBPool.ID
+	}
+
+	httpPath, err := eGW.buildProxiedEdgeEndpointURL(types.LBServerPoolPath + lbPoolConfig.ID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get Edge Gateway API endpoint: %s", err)
+	}
+
+	// Result should be 204, if not we expect an error of type types.NSXError
+	_, err = eGW.client.ExecuteRequestWithCustomError(httpPath, http.MethodPut, types.AnyXMLMime,
+		"error while updating load balancer server pool : %s", lbPoolConfig, &types.NSXError{})
+	if err != nil {
+		return nil, err
+	}
+
+	readPool, err := eGW.ReadLBServerPool(&types.LBPool{ID: lbPoolConfig.ID})
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve server pool with ID (%s) after update: %s", lbPoolConfig.ID, err)
+	}
+	return readPool, nil
+}
+
+// DeleteLBServerPool is able to delete the types.LBPool type by Name and/or ID.
+// If both - Name and ID are specified it performs a lookup by ID and returns an error if the specified name and found
+// name do not match.
+func (eGW *EdgeGateway) DeleteLBServerPool(lbPoolConfig *types.LBPool) error {
```

Comment:
Why need this struct if only `id` is used?

Reply:
Same as above. I've created these as flexible functions even for future. Having them it is very cheap and easy to create wrappers such as `DeleteLBServerPoolByID`, `DeleteLBServerPoolByName`, etc.

Reply:
Added for both - Server Pool and Service Monitor.